include/dfu/dfu_target_modem_full.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/** @file dfu_target_modem_full.h
+ *
+ * @defgroup dfu_target_modem_full Full Modem DFU Target
+ * @{
+ * @brief DFU Target full modem updates.
+ */
+
+#ifndef DFU_TARGET_MODEM_FULL_H__
+#define DFU_TARGET_MODEM_FULL_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <device.h>
+#include <dfu/dfu_target.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief Configure resources required by dfu_target_modem_full
+ *
+ * @param[in] buf Buffer to use for buffering streamed writes.
+ * @param[in] len Length of buf in bytes
+ * @param[in] dev Device to write the update to
+ * @param[in] dev_offset Offset within the device to store the update to
+ * @param[in] dev_size Total size of the device
+ *
+ * @retval non-negative integer if successful, otherwise negative errno.
+ */
+int dfu_target_modem_full_cfg(uint8_t *buf, size_t len,
```

Comment:
I feel like we will get burnt in the next round of review if I don't stick to the 'module_name _ something' naming convention.

Reply:
Ok, one final attempt to convince you :D

The name has three parts "dfu_target", "modem_full", "cfg"/"identify"/"init" etc. Changing "modem_full" to "full_modem" doesn't mess with the hierarchy IMO. AFAIU, "modem_delta" and "modem_full" don't actually share any more code than they do with "mcuboot", so in my head, there are three targets: "modem_delta", "modem_full", and "mcuboot". "full_modem" also matches better with how it's said: "full modem update".

Reply:
Ah, so you want to change all the "modem_full" to "full_modem" I thought you meant just for this function.

If we change to `dfu_target_full_modem` then I think we can keep the good old `dfu_target_modem` as it is, since the difference is better shown.

Reply:
Any feedback on this @oyvindronningstad  ?

Reply:
I slightly prefer modem_delta over modem to reduce ambiguity with full modem, but I'll leave it up to you.